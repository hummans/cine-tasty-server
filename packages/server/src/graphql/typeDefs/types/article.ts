import { gql } from 'apollo-server';

export default gql`
  type Article {
    publishedAt: String
    description: String
    content: String
    source: String
    author: String
    title: String
    image: String
    url: String
    id: ID
  }
`;
