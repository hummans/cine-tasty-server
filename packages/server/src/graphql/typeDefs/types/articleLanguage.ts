import { gql } from 'apollo-server';

export default gql`
  enum ArticleLanguage {
    AR
    DE
    EN
    ES
    FR
    HE
    IT
    NL
    NO
    PT
    RU
    SE
    UD
    ZH
  }
`;
