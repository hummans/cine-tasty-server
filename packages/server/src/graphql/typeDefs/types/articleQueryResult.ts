import { gql } from 'apollo-server';

export default gql`
  type ArticleQueryResult {
    items: [Article!]!
    hasMore: Boolean!
  }
`;
