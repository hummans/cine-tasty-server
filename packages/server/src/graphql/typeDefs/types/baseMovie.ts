import { gql } from 'apollo-server';

export default gql`
  type BaseMovie {
    original_title: String
    video: Boolean
    title: String
    adult: Boolean
    release_date: String
    backdrop_path: String
    genre_ids(language: ISO6391Language): [String!]!
    overview: String
    vote_average: Float
    media_type: String
    poster_path: String
    popularity: Float
    original_language: String
    vote_count: Float
    id: Int
  }
`;
