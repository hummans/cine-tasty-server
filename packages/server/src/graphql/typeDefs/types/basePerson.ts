import { gql } from 'apollo-server';

export default gql`
  type BasePerson {
    profile_path: String
    adult: Boolean
    id: Int
    popularity: Float
    known_for: [PersonKnowFor!]!
    name: String
  }
`;
