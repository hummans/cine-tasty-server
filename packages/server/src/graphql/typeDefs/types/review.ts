import { gql } from 'apollo-server';

export default gql`
  type Review {
    author: String
    content: String
    id: ID
    url: String
  }
`;
