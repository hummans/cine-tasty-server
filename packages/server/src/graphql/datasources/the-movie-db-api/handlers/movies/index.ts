import {
  BaseMovie,
  TrendingMoviesQueryResult,
  QueryMovieArgs,
  Movie,
  MovieReviewsArgs,
  SimilarMoviesQueryResult,
  ReviewsQueryResult,
  ReviewItem,
  MovieSimilarArgs,
  TrendingMoviesArgs,
} from '../../../../../lib/types';
import { GetTMDBApiRequest, TrendingMoviesEndpoints } from '../../../../../types';

const BASE_ENDPOINT = 'movie';

type BaseResponse = {
  total_results: number;
  total_pages: number;
  page: number;
};

type GetBaseMovieResponse = BaseResponse & {
  results: BaseMovie[];
};

type GetReviewsResponse = BaseResponse & {
  results: ReviewItem[];
};

type GetRequestParams = { page: number } | { append_to_response: string };

export interface Props {
  getSimilars(params: MovieSimilarArgs): Promise<SimilarMoviesQueryResult>;
  getReviews({ id, reviewsPage }: MovieReviewsArgs): Promise<ReviewsQueryResult>;
  getMovie: (params: QueryMovieArgs) => Promise<Movie | null>;
  getTrendingItem: (
    params: TrendingMoviesArgs,
    resource: TrendingMoviesEndpoints,
  ) => Promise<TrendingMoviesQueryResult>;
}

class MovieHandler implements Props {
  get: GetTMDBApiRequest;

  constructor(execGetRequest: GetTMDBApiRequest) {
    this.get = execGetRequest;
  }

  async getMovie({ id, language }: QueryMovieArgs): Promise<Movie | null> {
    const movie = await this.get<
      GetRequestParams,
      Promise<Movie & { status_message?: string }>
    >(`${BASE_ENDPOINT}/${id}`, { append_to_response: 'videos,credits' }, language);

    if (typeof movie.status_message === 'string') {
      return null;
    }

    return movie;
  }

  async getReviews({ id, reviewsPage }: MovieReviewsArgs): Promise<ReviewsQueryResult> {
    const { total_pages: totalPages, total_results, results } = await this.get<
      GetRequestParams,
      Promise<GetReviewsResponse>
    >(`${BASE_ENDPOINT}/${id}/reviews`, {
      page: reviewsPage,
    });

    return {
      hasMore: reviewsPage < totalPages,
      total_pages: totalPages,
      total_results,
      items: results,
    };
  }

  async getSimilars({
    id,
    similarsPage,
    language,
  }: MovieSimilarArgs): Promise<SimilarMoviesQueryResult> {
    const { total_pages: totalPages, total_results, results } = await this.get<
      GetRequestParams,
      Promise<GetBaseMovieResponse>
    >(
      `${BASE_ENDPOINT}/${id}/similar`,
      {
        page: similarsPage,
      },
      language,
    );

    return {
      hasMore: similarsPage < totalPages,
      total_pages: totalPages,
      total_results,
      items: results,
    };
  }

  async getTrendingItem(
    { page, language }: TrendingMoviesArgs,
    resource: TrendingMoviesEndpoints,
  ): Promise<TrendingMoviesQueryResult> {
    const { total_pages: totalPages, total_results, results } = await this.get<
      GetRequestParams,
      Promise<GetBaseMovieResponse>
    >(resource, { page }, language);

    return {
      hasMore: page < totalPages,
      total_pages: totalPages,
      total_results,
      items: results,
    };
  }
}

export default MovieHandler;
