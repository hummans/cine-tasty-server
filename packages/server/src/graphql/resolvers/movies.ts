import MediaGenresHandler from '../datasources/the-movie-db-api/handlers/media-genres';

import {
  QueryResolvers,
  TrendingMoviesQueryResult,
  TrendingMoviesNow_PlayingArgs as TrendingMoviesNowPlayingArgs,
  TrendingMoviesPopularArgs,
  TrendingMoviesTop_RatedArgs as TrendingMoviesTopRatedArgs,
  TrendingMoviesUpcomingArgs,
  QueryMovieArgs,
  BaseMovieGenre_IdsArgs as BaseMovieGenreIdsArgs,
  Movie,
  MovieGenresArgs,
  MediaType,
  CastItem,
  CrewItem,
  MediaVideo,
  MovieReviewsArgs,
  ReviewsQueryResultResolvers,
  SimilarMoviesQueryResultResolvers,
  MovieSimilarArgs,
} from '../../lib/types';
import { Context, MediaGenre, MediaItem, MediaCredits } from '../../types';

const mediaGenres = new MediaGenresHandler();

type MovieVideos = {
  results: MediaVideo[];
};

const BASE_VIDEO_THHUMBNAIL_URL = 'https://img.youtube.com/vi';

const resolvers: QueryResolvers = {
  Query: {
    trending_movies: (): {} => ({}),

    movie: (
      _: {},
      args: QueryMovieArgs,
      { dataSources }: Context,
    ): Promise<Movie | null> => dataSources.tmdb.getMovie(args),
  },

  BaseMovie: {
    genre_ids: (
      { genre_ids }: MediaItem,
      { language }: BaseMovieGenreIdsArgs,
    ): Promise<string[]> =>
      mediaGenres.getMediaGenres(genre_ids, MediaType.Movie.toLowerCase(), language),
  },

  Movie: {
    spoken_languages: ({
      spoken_languages,
    }: {
      spoken_languages: Array<{ name: string }>;
    }): string[] => spoken_languages.map(({ name }) => name),

    production_countries: ({
      production_countries,
    }: {
      production_countries: Array<{ name: string }>;
    }): string[] => production_countries.map(({ name }) => name),

    cast: ({ credits }: { credits: MediaCredits }): CastItem[] =>
      credits.cast.map(castItem => ({
        name: castItem.name,
        profile_path: castItem.profile_path,
        id: castItem.id,
        character: castItem.character,
      })),

    crew: ({ credits }: { credits: MediaCredits }): CrewItem[] =>
      credits.crew.map(castItem => ({
        profile_path: castItem.profile_path,
        department: castItem.department,
        id: castItem.id,
        job: castItem.job,
        name: castItem.name,
      })),

    reviews: (
      _: {},
      args: MovieReviewsArgs,
      { dataSources }: Context,
    ): ReviewsQueryResultResolvers => dataSources.tmdb.getMovieReviews(args),

    similar: (
      _: {},
      args: MovieSimilarArgs,
      { dataSources }: Context,
    ): SimilarMoviesQueryResultResolvers => dataSources.tmdb.getSimilarMovies(args),

    videos: ({ videos }: { videos: MovieVideos }): MediaVideo[] =>
      videos.results
        .filter(({ site }) => site === 'YouTube')
        .map(video => ({
          key: video.key,
          name: video.name,
          site: video.site,
          id: `${video.id}`,
          type: video.type,
          thumbnail: {
            extra_small: `${BASE_VIDEO_THHUMBNAIL_URL}/${video.key}/default.jpg`,
            small: `${BASE_VIDEO_THHUMBNAIL_URL}/${video.key}/mqdefault.jpg`,
            medium: `${BASE_VIDEO_THHUMBNAIL_URL}/${video.key}/hqdefault.jpg`,
            large: `${BASE_VIDEO_THHUMBNAIL_URL}/${video.key}/sddefault.jpg`,
            extra_large: `${BASE_VIDEO_THHUMBNAIL_URL}/${video.key}/maxresdefault.jpg`,
          },
        })),

    genres: (
      { genres }: { genres: Array<MediaGenre> },
      { language }: MovieGenresArgs,
    ): Promise<string[]> => {
      const genreIds = genres.map(({ id }) => id);

      return mediaGenres.getMediaGenres(
        genreIds,
        MediaType.Movie.toLowerCase(),
        language,
      );
    },
  },

  TrendingMovies: {
    now_playing: (
      _: {},
      { args }: TrendingMoviesNowPlayingArgs,
      { dataSources }: Context,
    ): Promise<TrendingMoviesQueryResult> =>
      dataSources.tmdb.getTrendingMoviesItem(args, 'now_playing'),

    popular: (
      _: {},
      { args }: TrendingMoviesPopularArgs,
      { dataSources }: Context,
    ): Promise<TrendingMoviesQueryResult> =>
      dataSources.tmdb.getTrendingMoviesItem(args, 'popular'),

    top_rated: (
      _: {},
      { args }: TrendingMoviesTopRatedArgs,
      { dataSources }: Context,
    ): Promise<TrendingMoviesQueryResult> =>
      dataSources.tmdb.getTrendingMoviesItem(args, 'top_rated'),

    upcoming: (
      _: {},
      { args }: TrendingMoviesUpcomingArgs,
      { dataSources }: Context,
    ): Promise<TrendingMoviesQueryResult> =>
      dataSources.tmdb.getTrendingMoviesItem(args, 'upcoming'),
  },
};

export default resolvers;
