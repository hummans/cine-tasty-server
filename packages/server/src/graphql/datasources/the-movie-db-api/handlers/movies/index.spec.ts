const mockRestDataSourceGet = jest.fn();

import { rawMovie, rawMovieDetail } from '../../../../../__tests__/mocks/movies.stub';
import { review } from '../../../../../__tests__/mocks/review.stub';
import { TrendingMoviesEndpoints } from '../../../../../types';
import { Iso6391Language } from '../../../../../lib/types';
import CONSTANTS from '../../utils/constants';
import MovieHandler from '.';

describe('Unity: MovieHandler', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('getTrendingItem()', () => {
    it('should return an array of the now playing movies from the TheMovieDB API', async () => {
      mockRestDataSourceGet.mockReturnValueOnce({
        total_pages: 1,
        total_results: 1,
        results: [rawMovie],
      });

      const movieHandler = new MovieHandler(mockRestDataSourceGet);

      const result = await movieHandler.getTrendingItem(
        {
          page: 1,
          language: Iso6391Language.Ptbr,
        },
        TrendingMoviesEndpoints.NowPlaying,
      );

      expect(mockRestDataSourceGet).toHaveBeenCalledWith(
        TrendingMoviesEndpoints.NowPlaying,
        { page: 1 },
        Iso6391Language.Ptbr,
      );
      expect(mockRestDataSourceGet.mock.calls.length).toBe(1);
      expect(result.items).toMatchSnapshot();
      expect(result.hasMore).toEqual(false);
      expect(result.total_pages).toEqual(1);
      expect(result.total_results).toEqual(1);
    });

    it('should return an array of the popular movies from the TheMovieDB API', async () => {
      mockRestDataSourceGet.mockReturnValueOnce({
        total_pages: 1,
        total_results: 1,
        results: [rawMovie],
      });

      const movieHandler = new MovieHandler(mockRestDataSourceGet);

      const result = await movieHandler.getTrendingItem(
        {
          page: 1,
          language: Iso6391Language.Ptbr,
        },
        TrendingMoviesEndpoints.Popular,
      );

      expect(mockRestDataSourceGet).toHaveBeenCalledWith(
        TrendingMoviesEndpoints.Popular,
        { page: 1 },
        Iso6391Language.Ptbr,
      );
      expect(mockRestDataSourceGet.mock.calls.length).toBe(1);
      expect(result.items).toMatchSnapshot();
      expect(result.hasMore).toEqual(false);
      expect(result.total_pages).toEqual(1);
      expect(result.total_results).toEqual(1);
    });

    it('should return an array of the top rated movies from the TheMovieDB API', async () => {
      mockRestDataSourceGet.mockReturnValueOnce({
        total_pages: 1,
        total_results: 1,
        results: [rawMovie],
      });

      const movieHandler = new MovieHandler(mockRestDataSourceGet);

      const result = await movieHandler.getTrendingItem(
        {
          page: 1,
          language: Iso6391Language.Ptbr,
        },
        TrendingMoviesEndpoints.TopRated,
      );

      expect(mockRestDataSourceGet).toHaveBeenCalledWith(
        TrendingMoviesEndpoints.TopRated,
        { page: 1 },
        Iso6391Language.Ptbr,
      );
      expect(mockRestDataSourceGet.mock.calls.length).toBe(1);
      expect(result.items).toMatchSnapshot();
      expect(result.hasMore).toEqual(false);
      expect(result.total_pages).toEqual(1);
      expect(result.total_results).toEqual(1);
    });

    it('should return an array of the upcoming movies from the TheMovieDB API', async () => {
      mockRestDataSourceGet.mockReturnValueOnce({
        total_pages: 1,
        total_results: 1,
        results: [rawMovie],
      });

      const movieHandler = new MovieHandler(mockRestDataSourceGet);

      const result = await movieHandler.getTrendingItem(
        {
          page: 1,
          language: Iso6391Language.Ptbr,
        },
        TrendingMoviesEndpoints.Upcoming,
      );

      expect(mockRestDataSourceGet).toHaveBeenCalledWith(
        TrendingMoviesEndpoints.Upcoming,
        { page: 1 },
        Iso6391Language.Ptbr,
      );
      expect(mockRestDataSourceGet.mock.calls.length).toBe(1);
      expect(result.items).toMatchSnapshot();
      expect(result.hasMore).toEqual(false);
      expect(result.total_pages).toEqual(1);
      expect(result.total_results).toEqual(1);
    });
  });

  describe('getMovie()', () => {
    it('should get the details of a movie with certain id from TheMovideDB API', async () => {
      mockRestDataSourceGet.mockReturnValueOnce(rawMovieDetail);

      const movieHandler = new MovieHandler(mockRestDataSourceGet);

      const result = await movieHandler.getMovie({
        id: '1',
        language: Iso6391Language.Ptbr,
      });

      expect(mockRestDataSourceGet).toHaveBeenCalledWith(
        'movie/1',
        {
          append_to_response: 'videos,credits',
        },
        'PTBR',
      );

      expect(mockRestDataSourceGet.mock.calls.length).toBe(1);

      expect(result).toMatchSnapshot();
    });

    it("should return null when the movie doesn't exist", async () => {
      mockRestDataSourceGet.mockReturnValueOnce({
        status_code: CONSTANTS.TMDBAPI_ITEM_NOT_FOUND_CODE,
      });

      const movieHandler = new MovieHandler(mockRestDataSourceGet);

      const result = await movieHandler.getMovie({
        id: '1',
        language: Iso6391Language.Ptbr,
      });

      expect(mockRestDataSourceGet).toHaveBeenCalledWith(
        'movie/1',
        {
          append_to_response: 'videos,credits',
        },
        'PTBR',
      );

      expect(mockRestDataSourceGet.mock.calls.length).toBe(1);

      expect(result).toBeNull();
    });
  });

  describe('getSimilars()', () => {
    it('should get similar movies of a movie with certain id from TheMovideDB API', async () => {
      mockRestDataSourceGet.mockReturnValueOnce({
        id: 1,
        page: 1,
        results: [rawMovie],
        total_pages: 1,
        total_results: 1,
      });

      const movieHandler = new MovieHandler(mockRestDataSourceGet);

      const result = await movieHandler.getSimilars({
        id: '1',
        similarsPage: 1,
        language: Iso6391Language.Ptbr,
      });

      expect(mockRestDataSourceGet).toHaveBeenCalledWith(
        'movie/1/similar',
        {
          page: 1,
        },
        'PTBR',
      );

      expect(mockRestDataSourceGet.mock.calls.length).toBe(1);

      expect(result).toEqual({
        hasMore: false,
        total_pages: 1,
        total_results: 1,
        items: [rawMovie],
      });

      expect(result.hasMore).toEqual(false);
    });

    it('should get similar movies of a movie with certain id from TheMovideDB API and return hasMore as true when has more items to be paginated', async () => {
      mockRestDataSourceGet.mockReturnValueOnce({
        id: 1,
        page: 1,
        results: [rawMovie],
        total_pages: 2,
        total_results: 2,
      });

      const movieHandler = new MovieHandler(mockRestDataSourceGet);

      const result = await movieHandler.getSimilars({
        id: '1',
        similarsPage: 1,
        language: Iso6391Language.Ptbr,
      });

      expect(mockRestDataSourceGet).toHaveBeenCalledWith(
        'movie/1/similar',
        {
          page: 1,
        },
        'PTBR',
      );

      expect(mockRestDataSourceGet.mock.calls.length).toBe(1);

      expect(result).toEqual({
        hasMore: true,
        total_pages: 2,
        total_results: 2,
        items: [rawMovie],
      });

      expect(result.hasMore).toEqual(true);
    });
  });

  describe('getReviews()', () => {
    it('should get the reviews of a movie with certain id from TheMovideDB API', async () => {
      mockRestDataSourceGet.mockReturnValueOnce({
        id: 1,
        page: 1,
        results: [review],
        total_pages: 1,
        total_results: 1,
      });

      const movieHandler = new MovieHandler(mockRestDataSourceGet);

      const result = await movieHandler.getReviews({
        id: '1',
        reviewsPage: 1,
        language: Iso6391Language.Ptbr,
      });

      expect(mockRestDataSourceGet).toHaveBeenCalledWith(
        'movie/1/reviews',
        { page: 1 },
        Iso6391Language.Ptbr,
      );

      expect(mockRestDataSourceGet.mock.calls.length).toBe(1);

      expect(result).toEqual({
        hasMore: false,
        total_pages: 1,
        total_results: 1,
        items: [review],
      });

      expect(result.hasMore).toEqual(false);
    });

    it('should get the reviews of a movie with certain id from TheMovideDB API and return hasMore as true when has more items to be paginated', async () => {
      mockRestDataSourceGet.mockReturnValueOnce({
        id: 1,
        page: 1,
        results: [review],
        total_pages: 2,
        total_results: 2,
      });

      const movieHandler = new MovieHandler(mockRestDataSourceGet);

      const result = await movieHandler.getReviews({
        id: '1',
        reviewsPage: 1,
        language: Iso6391Language.Ptbr,
      });

      expect(mockRestDataSourceGet).toHaveBeenCalledWith(
        'movie/1/reviews',
        { page: 1 },
        Iso6391Language.Ptbr,
      );

      expect(mockRestDataSourceGet.mock.calls.length).toBe(1);

      expect(result).toEqual({
        hasMore: true,
        total_pages: 2,
        total_results: 2,
        items: [review],
      });

      expect(result.hasMore).toEqual(true);
    });
  });
});
