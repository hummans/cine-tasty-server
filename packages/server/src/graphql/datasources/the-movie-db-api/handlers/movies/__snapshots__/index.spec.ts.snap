// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MovieHandler] should get the details of a movie with certain id from TheMovideDB API 1`] = `
Object {
  "adult": false,
  "backdrop_path": "/backdrop_path.jpg",
  "budget": 123,
  "credits": Object {
    "cast": Array [
      Object {
        "cast_id": 4,
        "character": "character",
        "credit_id": "credit_id",
        "gender": 1,
        "id": 1,
        "name": "name",
        "order": 0,
        "profile_path": "/profile_path.jpg",
      },
    ],
    "crew": Array [
      Object {
        "credit_id": "credit_id",
        "department": "department",
        "gender": 2,
        "id": 2,
        "job": "job",
        "name": "Dname",
        "profile_path": "profile_path",
      },
    ],
  },
  "genres": Array [
    Object {
      "id": 2,
      "name": "Movie Genre",
    },
  ],
  "homepage": "homepage",
  "id": 123,
  "original_language": "original_language",
  "original_title": "original_title",
  "overview": "overview",
  "popularity": 1,
  "poster_path": "poster_path",
  "production_companies": Array [
    Object {
      "id": 123,
      "logo_path": "/logo_path.png",
      "name": "name",
      "origin_country": "origin_country",
    },
  ],
  "production_countries": Array [
    Object {
      "iso_3166_1": "iso_3166_1",
      "name": "name",
    },
  ],
  "release_date": "release_date",
  "revenue": 123,
  "runtime": 123,
  "spoken_languages": Array [
    Object {
      "iso_639_1": "iso_639_1",
      "name": "name",
    },
  ],
  "status": "status",
  "tagline": "tagline",
  "title": "title",
  "video": false,
  "videos": Object {
    "results": Array [
      Object {
        "id": "id",
        "iso_3166_1": "iso_3166_1",
        "iso_639_1": "iso_639_1",
        "key": "key",
        "name": "name",
        "site": "YouTube",
        "size": 1080,
        "type": "type",
      },
      Object {
        "id": "id",
        "iso_3166_1": "iso_3166_1",
        "iso_639_1": "iso_639_1",
        "key": "key",
        "name": "name",
        "site": "Other",
        "size": 1080,
        "type": "type",
      },
    ],
  },
  "vote_average": 1.8,
  "vote_count": 3439,
}
`;

exports[`[MovieHandler] should return an array of the now playing movies from the TheMovieDB API 1`] = `
Array [
  Object {
    "adult": false,
    "backdrop_path": "/backdrop_path.jpg",
    "genre_ids": Array [
      2,
    ],
    "id": 530915,
    "original_language": "original_language",
    "original_title": "original_title",
    "overview": "overview",
    "popularity": 222.017,
    "poster_path": "/poster_path.jpg",
    "release_date": "release_date",
    "title": "title",
    "video": false,
    "vote_average": 8.1,
    "vote_count": 1530,
  },
]
`;

exports[`[MovieHandler] should return an array of the popular movies from the TheMovieDB API 1`] = `
Array [
  Object {
    "adult": false,
    "backdrop_path": "/backdrop_path.jpg",
    "genre_ids": Array [
      2,
    ],
    "id": 530915,
    "original_language": "original_language",
    "original_title": "original_title",
    "overview": "overview",
    "popularity": 222.017,
    "poster_path": "/poster_path.jpg",
    "release_date": "release_date",
    "title": "title",
    "video": false,
    "vote_average": 8.1,
    "vote_count": 1530,
  },
]
`;

exports[`[MovieHandler] should return an array of the top rated movies from the TheMovieDB API 1`] = `
Array [
  Object {
    "adult": false,
    "backdrop_path": "/backdrop_path.jpg",
    "genre_ids": Array [
      2,
    ],
    "id": 530915,
    "original_language": "original_language",
    "original_title": "original_title",
    "overview": "overview",
    "popularity": 222.017,
    "poster_path": "/poster_path.jpg",
    "release_date": "release_date",
    "title": "title",
    "video": false,
    "vote_average": 8.1,
    "vote_count": 1530,
  },
]
`;

exports[`[MovieHandler] should return an array of the upcoming movies from the TheMovieDB API 1`] = `
Array [
  Object {
    "adult": false,
    "backdrop_path": "/backdrop_path.jpg",
    "genre_ids": Array [
      2,
    ],
    "id": 530915,
    "original_language": "original_language",
    "original_title": "original_title",
    "overview": "overview",
    "popularity": 222.017,
    "poster_path": "/poster_path.jpg",
    "release_date": "release_date",
    "title": "title",
    "video": false,
    "vote_average": 8.1,
    "vote_count": 1530,
  },
]
`;
